src/core/json/json_like.rs

```diff
@@ -41,6 +44,29 @@ pub trait JsonLike<'json>: Sized {
     fn get_path<T: AsRef<str>>(&'json self, path: &[T]) -> Option<&Self>;
     fn get_key(&'json self, path: &str) -> Option<&Self>;
     fn group_by(&'json self, path: &[String]) -> HashMap<String, Vec<&Self>>;
+
+    fn map<Err>(self, mut mapper: impl FnMut(Self) -> Result<Self, Err>) -> Result<Self, Err> {
```

Comment:
I think this method solves a very specific problem in a very specific way and doesn't need to be placed in JsonLike.

Reply:
I moved this to another trait since it could re reused in other places later e.g. jit batching